### ISSUE TITLE
Address component to display user-owned accounts via their client names 

### ISSUE THREAD

**Issue by eriknson** [MetaMask/snaps#2758]
Let's parse a user's own accounts to be represented via their display names (e.g. "Gaming Account 1") instead of the truncated address (e.g. "0xEqT...83x") in our accounts component.

[Figma](https://www.figma.com/design/gHWl7G8ih3YXesly2XaA0R/Snaps-DS?node-id=329-1524&t=yPhqvbM20tFUWtv8-1)

**Comment by danroc** [comment]
I think we should add a requirement for the address to be a CAIP-10 address. We may also need the chain ID information if we decide to customize the address display per chain.

**Comment by danroc** [comment]
The `accountId` is a good addition if account Snaps will use this component to display managed accounts, as it's the preferred way to identify managed accounts within MetaMask and account Snaps.

**Comment by FrederikBolding** [comment]
> I think we should add a requirement for the address to be a CAIP-10 address. We may also need the chain ID information if we decide to customize the address display per chain.

The `address` prop already supports CAIP-10 addresses, so we should be good on that front.

> The `accountId` is a good addition if account Snaps will use this component to display managed accounts, as it's the preferred way to identify managed accounts within MetaMask and account Snaps.

Curious if it wouldn't be better for the Snap to explicitly choose the CAIP-10 address it wants to display, instead of us maintaining the logic to choose that based on an account ID. At least that seems to match better what we were planning for `AccountSelector` etc. WDYT?

**Comment by danroc** [comment]
> Curious if it wouldn't be better for the Snap to explicitly choose the CAIP-10 address it wants to display, instead of us maintaining the logic to choose that based on an account ID. At least that seems to match better what we were planning for AccountSelector etc. WDYT?

Agreed, it makes sense for the _Address_ component to expect an address. I think I confused it with the _Account_ component, which could have an `accountId` prop.
